From 9acdc9238a248a0d2e4b3aa71d5b11149fef8f01 Mon Sep 17 00:00:00 2001
From: Vlad Pelin <vlad.pelin@nxp.com>
Date: Tue, 6 Dec 2022 15:34:38 +0200
Subject: [PATCH 7/7] pkcs: guard EC key format with #ifdef

the G2 HSE interface differs from the G3 HSE interface;
because of this, the EC key format field is not present
in G2 versions of the interface, which requires a check
of HSE_PLATFORM to identify the interface version

Issue: ALB-9590
Upstream-Status: Pending 

Signed-off-by: Vlad Pelin <vlad.pelin@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 libpkcs/pkcs11_object.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/libpkcs/pkcs11_object.c b/libpkcs/pkcs11_object.c
index cd147fe..284b79d 100644
--- a/libpkcs/pkcs11_object.c
+++ b/libpkcs/pkcs11_object.c
@@ -296,7 +296,9 @@ CK_DEFINE_FUNCTION(CK_RV, C_CreateObject)(
 			import_key_req->keyLen[0] = pkey0_len;
 			import_key_req->keyLen[1] = 0u;
 			import_key_req->keyLen[2] = 0u;
+#if (HSE_PLATFORM == HSE_S32G3XX)
 			import_key_req->keyFormat.eccKeyFormat = HSE_KEY_FORMAT_ECC_PUB_RAW;
+#endif
 
 			if (key->key_class == CKO_PRIVATE_KEY) {
 
-- 
2.17.1

